### PR TITLE
SAK-32746 - Sometimes the anonymous return from SAK-32738 is null

### PR DESCRIPTION
**Pull request by jonespm**

### CODE REVIEW

File: portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java [hunk 1]
```diff
@@ -1051,7 +1051,7 @@ public PortalRenderContext startPageContext(String siteType, String title,
 
 				
 		User currentUser = UserDirectoryService.getCurrentUser();
-		Role role = site.getUserRole(currentUser.getId());
+		Role role = currentUser != null ? site.getUserRole(currentUser.getId()) : null;
```
Comment (ottenhoff) [inline portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java]: Is the site guaranteed not null?

**Comment by jonespm** [comment]
Really, I bet that's what it is, the site is for sure possible to be null. Sometimes I hate java, thanks for the check.

**Comment by jonespm** [comment]
Okay updated to protect currentUser and site, role is already null expected and protected below.

**Comment by buckett** [comment]
Unless the component manager has failed to startup I don't think it's possible for the user to be `null` as if there isn't a user the anonymous user will be returned and although the anonymous user doesn't have lots of values it does have an ID.

**Comment by jonespm** [comment]
Yeah, I'm guessing it was the site being null in this case (which is passed in as a null from some portal places), I just left the check in there for both to be extra safe. The error this caused seemed to not cause any  issue to the user but just printed up tons of stack traces to the log which looks better now.

**Pull request merged**